pkg/updater/gcs.go

```diff
@@ -277,7 +277,8 @@ func convertResult(log logrus.FieldLogger, nameCfg nameConfig, id string, header
 	// Append each result into the column
 	for _, suite := range result.suites {
 		for _, r := range flattenResults(suite.Suites.Suites...) {
-			if r.Skipped != nil && r.Skipped.Value == "" {
+			// "skipped" is the string that is always appended when the test is skipped without any reason in Ginkgo V2, e.g., "focus" is specified, and the test is skipped.
+			if r.Skipped != nil && (r.Skipped.Value == "" && (r.Skipped.Message == "skipped" || r.Skipped.Message == "")) {
```

Comment:
In v2:  `r.Skipped.Value` should be always empty.

In v1:  `r.Skipped.Message` should be always empty

Reply:
yes

Reply:
we do want to report the skipped ones, for ginkgo v2 we should not continue if r.Skipped != nil and r.Skipped.Message != ""

Reply:
manage to repro now https://go.dev/play/p/wYK_PJ3QZtS?v=goprev , this is easier to understand

Reply:
based on your finding Chen, this was the PR introducing the new format https://github.com/GoogleCloudPlatform/testgrid/pull/832/

that added a Value `chardata` , but my reading is that Skipped never has such field

https://llg.cubic.org/docs/junit/

```
      <!-- skipped can appear 0 or once. optional -->
      <skipped message=""   <!-- message/description string why the test case was skipped. optional -->
	  />

      <!-- error indicates that the test errored.
           An errored test had an unanticipated problem.
           For example an unchecked throwable (exception), crash or a problem with the implementation of the test.
           Contains as a text node relevant data for the error, for example a stack trace. optional -->
      <error message="" <!-- The error message. e.g., if a java exception is thrown, the return value of getMessage() -->
	     type=""    <!-- The type of error that occured. e.g., if a java execption is thrown the full class name of the exception. -->
	     >error description</error>

      <!-- failure indicates that the test failed.
           A failure is a condition which the code has explicitly failed by using the mechanisms for that purpose.
           For example via an assertEquals.
           Contains as a text node relevant data for the failure, e.g., a stack trace. optional -->
      <failure message="" <!-- The message specified in the assert. -->
	       type=""    <!-- The type of the assert. -->
	       >failure description</failure>

```

I think that we should fix the field removing the value and only continue if

```
	if r.Skipped != nil && r.Skipped.Message == "" {
				continue
			}
```

or always report the skipped entry

Reply:
cc@ @michelle192837 @chizhg

Reply:
> we do want to report the skipped ones, for ginkgo v2 we should not continue if r.Skipped != nil and r.Skipped.Message != ""

this won't be skipped, I think the testcases I added has already verified this.

Reply:
> I think that we should fix the field removing the value

I don't think so, the `value` field is still needed for backward compatibility, iiuc, the `value` here is the unmarshalled from v1 `Message` field.

```
type JUnitSkipped struct {
	Message string `xml:",chardata"`
}
```

testgrid should support both the old and new format.

Reply:
fair enough to me, I just said that because this moved from 

```
	Skipped    *string     `xml:"skipped,omitempty"`
```

to 

```
// Skipped holds <skipped/> elements.
type Skipped struct {
	Message string `xml:"message,attr"`
	Value   string `xml:",chardata"`
}
```

I let you decide, I don't have any strong opinion, just to keep the reports fully working again ;)

Reply:
> or always report the skipped entry

k8s now has more than 7k+ e2e testcases, show all the skipped testcase with no useful information doesn't bring us something. 

thinking of the case that I run e2e and only focus on 1 testcase, we will only have only one valid result but all others are skipped, do we really all those skipped ones?

example of the skipped one in v2
```
          <testcase name="[It] [sig-storage] In-tree Volumes [Driver: azure-disk] [Testpattern: Pre-provisioned PV (default fs)] subPath should support file as subpath [LinuxOnly]" classname="Kubernetes e2e suite" status="skipped" time="0">
              <skipped message="skipped"></skipped>
          </testcase>
```


I think this is why testgrid doesn't show the skipped testcase which has the empty `value`.

```golang
if r.Skipped != nil && r.Skipped.Value == ""
```
and then treat it as invisible.

in v1,

```
<testcase name="[sig-storage] In-tree Volumes [Driver: nfs] [Testpattern: Dynamic PV (block volmode)] multiVolume [Slow] should access to two volumes with different volume mode and retain data across pod recreation on different node" classname="Kubernetes e2e suite" time="0">
<skipped></skipped>
</testcase>
```

I am new to this project, so all of those are just based on the assumption, so pls correct me if I miss something.

Reply:
ah, no, you are right, that should be the reason then and an I fully agree with that, we should not report the empty ones

Reply:
I agree with suppressing the "empty ones". But note that ginkgo v2 no longer skips them with an empty string. Instead it uses "skipped" as message. For example, when I run e2e.test with focus on a single test, I get for the others:
```
          <testcase name="[It] [sig-storage] In-tree Volumes [Driver: emptydir] [Testpattern: Dynamic PV (ntfs)][Feature:Windows] volumes should allow exec of files on the volume" classname="Kubernetes e2e suite" status="skipped" time="0">
              <skipped message="skipped"></skipped>
          </testcase>
```

We should also ignore those.

The interesting ones are the ones which were skipped at runtime. They tend to have a more useful explanation that IMHO is worthwhile showing in testgrid because a) running these tests cost resources and b) perhaps they were supposed to be running.

Reply:
Never mind, it looks like the code already handles "message=skipped". Looks good to me.

Reply:
(As an aside, thanks for clarifying why empty skipped cases aren't reported, I knew that was a thing and had forgotten why).

Reply:
This is a long thread and I haven't checked what it is all about, but there has been a Google-internal issue about TestGrid not showing skipped tests, and the bug number is 221515005